### ISSUE TITLE
build-error with swift 07-25

### ISSUE THREAD

**Issue by tnobody** [OpenKitten/MongoKitten#35]
Hi,

i would like to build MongoKitten. Unfortunately even in an plane project i got following build output:
```bash 
Cloning https://github.com/PlanTeam/MongoKitten.git
HEAD is now at 4df9cd0 Merge pull request #29 from LoganWright/patch-1
Resolved version: 1.4.1
Cloning https://github.com/CryptoKitten/MD5.git
HEAD is now at 3ac877d Newer CryptoEssentials
Resolved version: 0.10.0
Cloning https://github.com/CryptoKitten/CryptoEssentials.git
HEAD is now at 7165379 Merge pull request #12 from CryptoKitten/07-25
Resolved version: 0.10.0
Cloning https://github.com/CryptoKitten/SCRAM.git
HEAD is now at 5f3f89d Updated to 07-25
Resolved version: 0.10.0
Cloning https://github.com/CryptoKitten/PBKDF2.git
HEAD is now at 1c26567 Updated to 07-25
Resolved version: 0.10.0
Cloning https://github.com/CryptoKitten/HMAC.git
HEAD is now at a748975 Merge pull request #7 from CryptoKitten/07-25
Resolved version: 0.10.0
Cloning https://github.com/CryptoKitten/SHA1.git
HEAD is now at 2ea8e6c Merge pull request #5 from CryptoKitten/07-25
Resolved version: 0.10.0
Cloning https://github.com/OpenKitten/BSON.git
HEAD is now at 2c85f41 Fix for ISO8601DateFormatter change
Resolved version: 3.5.1
Cloning https://github.com/open-swift/C7.git
HEAD is now at ec599fb Updated to Swift 07-25 (#50)
Resolved version: 0.10.0
Cloning https://github.com/czechboy0/Socks.git
HEAD is now at 05154cb Updated to 07-25 (#67)
Resolved version: 0.10.0
Cloning https://github.com/ketzusaka/Strand.git
HEAD is now at 6278d10 Merge pull request #8 from ketzusaka/07-20
Resolved version: 1.6.0
Compile Swift Module 'SocksCore' (14 sources)
Compile Swift Module 'Strand' (1 sources)
Compile Swift Module 'C7' (23 sources)
Compile Swift Module 'CryptoEssentials' (15 sources)
Compile Swift Module 'BSON' (12 sources)
Compile Swift Module 'Socks' (5 sources)
Compile Swift Module 'SocksCoreExampleTCPServer' (1 sources)
Compile Swift Module 'SocksCoreExampleTCPClient' (1 sources)
Compile Swift Module 'SocksExampleUDPClient' (1 sources)
Compile Swift Module 'SocksExampleUDPServer' (1 sources)
Compile Swift Module 'SocksExampleTCPServer' (1 sources)
Compile Swift Module 'SocksExampleTCPClient' (1 sources)
Compile Swift Module 'SocksCoreExampleTCPKeepAliveServer' (1 sources)
Linking ./.build/debug/SocksCoreExampleTCPClient
Linking ./.build/debug/SocksCoreExampleTCPServer
Linking ./.build/debug/SocksExampleTCPClient
Linking ./.build/debug/SocksExampleUDPServer
Linking ./.build/debug/SocksExampleUDPClient
Linking ./.build/debug/SocksExampleTCPServer
Linking ./.build/debug/SocksCoreExampleTCPKeepAliveServer
Compile Swift Module 'SHA1' (1 sources)
Compile Swift Module 'HMAC' (1 sources)
Compile Swift Module 'MD5' (1 sources)
Compile Swift Module 'PBKDF2' (1 sources)
/Users/timkeiner/Projects/swift/vapor/test-mongo/Packages/BSON-3.5.1/Sources/ExtendedJSON.swift:405:36: error: initializer for conditional binding must have Optional type, not 'Date'
                                if let date = fmt.date(from: dateString) {
                                   ^
<unknown>:0: error: build had 1 command failures
error: exit(1): /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2016-07-25-a.xctoolchain/usr/bin/swift-build-tool -f /Users/timkeiner/Projects/swift/vapor/test-mongo/.build/debug.yaml
````

I build with the 07-25 Toolchain
``` bash
 /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2016-07-25-a.xctoolchain/usr/bin/swift build
```

and had a clean project. With the following package.swift
```swift
import PackageDescription

let package = Package(
    name: "test-mongo",
    dependencies: [
      .Package(url: "https://github.com/PlanTeam/MongoKitten.git", majorVersion: 1, minor: 4)
    ]
)
```

OS X Version is: 10.11.16

Thanks in advance :)

**Comment by Obbut** [comment]
We are aware of this issue. We haven't found/implemented a Linux and macOS 10.11 compatible way of parsing ISO8601 dates for BSON, which is needed for the extended JSON parsing functionality.

I'll either look for a solution tomorrow or temporarily disable the date parsing to allow using MongoKitten on all operating systems.

**Workaround**: In the meantime, although this solution is a bit dirty, you can replace lines 400-412 of the file `ExtendedJSON.swift` in your working copy of BSON with `break Parser`. You can find your working copy of this file under your package in `Packages/BSON-3.5.1/Sources/ExtendedJSON.swift`.

**Comment by Joannis** [comment]
@tnobody I'll be working to fix this in the next version. @Obbut 's workaround is okay for now (if you're not using MongoDB Extended JSON with dated). But I'll add a cross-platform parser.

**Comment by Joannis** [comment]
Fixed in the newest BSON. Please fetch the dependencies again to solve the issue. And let us know any other issues!

**Comment by tnobody** [comment]
Hi guys thx for this quick response :)

But I encountered another issue :(

``` bash 
Cloning https://github.com/PlanTeam/MongoKitten.git
HEAD is now at 4df9cd0 Merge pull request #29 from LoganWright/patch-1
Resolved version: 1.4.1
Cloning https://github.com/CryptoKitten/MD5.git
HEAD is now at 3ac877d Newer CryptoEssentials
Resolved version: 0.10.0
Cloning https://github.com/CryptoKitten/CryptoEssentials.git
HEAD is now at 7165379 Merge pull request #12 from CryptoKitten/07-25
Resolved version: 0.10.0
Cloning https://github.com/CryptoKitten/SCRAM.git
HEAD is now at 5f3f89d Updated to 07-25
Resolved version: 0.10.0
Cloning https://github.com/CryptoKitten/PBKDF2.git
HEAD is now at 1c26567 Updated to 07-25
Resolved version: 0.10.0
Cloning https://github.com/CryptoKitten/HMAC.git
HEAD is now at a748975 Merge pull request #7 from CryptoKitten/07-25
Resolved version: 0.10.0
Cloning https://github.com/CryptoKitten/SHA1.git
HEAD is now at 2ea8e6c Merge pull request #5 from CryptoKitten/07-25
Resolved version: 0.10.0
Cloning https://github.com/OpenKitten/BSON.git
HEAD is now at a6ad594 Fixed a tests
Resolved version: 3.5.3
Cloning https://github.com/open-swift/C7.git
HEAD is now at ec599fb Updated to Swift 07-25 (#50)
Resolved version: 0.10.0
Cloning https://github.com/czechboy0/Socks.git
HEAD is now at 05154cb Updated to 07-25 (#67)
Resolved version: 0.10.0
Cloning https://github.com/ketzusaka/Strand.git
HEAD is now at 6278d10 Merge pull request #8 from ketzusaka/07-20
Resolved version: 1.6.0
Compile Swift Module 'SocksCore' (14 sources)
Compile Swift Module 'Strand' (1 sources)
Compile Swift Module 'C7' (23 sources)
Compile Swift Module 'CryptoEssentials' (15 sources)
Compile Swift Module 'Socks' (5 sources)
Compile Swift Module 'SocksCoreExampleTCPServer' (1 sources)
Compile Swift Module 'SocksCoreExampleTCPClient' (1 sources)
Compile Swift Module 'SocksExampleUDPServer' (1 sources)
Compile Swift Module 'SocksExampleUDPClient' (1 sources)
Compile Swift Module 'SocksExampleTCPServer' (1 sources)
Compile Swift Module 'SocksExampleTCPClient' (1 sources)
Compile Swift Module 'SocksCoreExampleTCPKeepAliveServer' (1 sources)
Compile Swift Module 'BSON' (12 sources)
Linking ./.build/debug/SocksCoreExampleTCPClient
Linking ./.build/debug/SocksCoreExampleTCPServer
Linking ./.build/debug/SocksExampleTCPClient
Linking ./.build/debug/SocksExampleUDPClient
Linking ./.build/debug/SocksExampleUDPServer
Linking ./.build/debug/SocksExampleTCPServer
Linking ./.build/debug/SocksCoreExampleTCPKeepAliveServer
Compile Swift Module 'SHA1' (1 sources)
Compile Swift Module 'HMAC' (1 sources)
Compile Swift Module 'MD5' (1 sources)
Compile Swift Module 'PBKDF2' (1 sources)
Compile Swift Module 'SCRAM' (1 sources)
/Users/timkeiner/Projects/swift/vapor/test-mongo/Packages/SCRAM-0.10.0/Sources/SCRAM.swift:41:31: warning: expected 'let' in conditional
        if let nonce = nonce, iterations = iterations, salt = salt {
                              ^
                              let 
/Users/timkeiner/Projects/swift/vapor/test-mongo/Packages/SCRAM-0.10.0/Sources/SCRAM.swift:41:56: warning: expected 'let' in conditional
        if let nonce = nonce, iterations = iterations, salt = salt {
                                                       ^
                                                       let 
Compile Swift Module 'MongoKitten' (13 sources)
Compile Swift Module 'test_mongo' (1 sources)

```

I just did:

```bash 
Tims-MBP:test-mongo timkeiner$ /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2016-07-25-a.xctoolchain/usr/bin/swift build --clean=dist
Tims-MBP:test-mongo timkeiner$ /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2016-07-25-a.xctoolchain/usr/bin/swift build
```

Setup as described  above

Thanks in Advance

**Comment by Obbut** [comment]
What is the issue, then? The output you pasted seems fine.

The two warnings will be solved soon but won't stop you from using MongoKitte.
